File: src/HA2/A3/Inbox.java
package src.HA2.A3;

public class Inbox {

    private Mail[] mails;


    public Inbox(int size) {
        mails = new Mail[size];
    }

    public void getHeaders(){
        for (Mail mail : mails) {
            System.out.println(mail.getRead() + "|" + mail.getSubject() + "|" + mail.getSender() + "|" + mail.getDatetime());
        }
    }

    public void open(int index){
        if(index >= mails.length){
            System.out.println("Index is too large!");
        }
        System.out.println(mails[index].getRead() + "|" + mails[index].getSubject() + "|" + mails[index].getSender() + "|" +mails[index].getDatetime());
        System.out.println(mails[index].getMessage());
        mails[index].isRead();
    }

    public int countUnread(){
        int count =0;
        for (Mail mail : mails) {
            if (!mail.getRead()) {
                count++;
            }
        }
        return count;
    }

    public Mail[] getMails() {
        return mails;
    }

    public void setMails(Mail[] mails) {
        this.mails = mails;
    }

    public void addMail(Mail m){
        for (int i = 0; i< mails.length; i++){
            if(mails[i] ==null){
                mails[i] = m;
                break;
            }
        }
    }

    public Mail getMail(int index){
        if (index >= 0 && index < mails.length) {
            return mails[index];
        }
        return null;
    }
}

File: python_HAs/HA10/A3/Inbox.py
class Inbox:
    def __init__(self):
        self.mails = []

    def add_mail(self, mail):
        self.mails.append(mail)

    def print_headers(self):
        for mail in self.mails:
            formatted_time = mail.datetime.strftime("%d.%m.%Y, %H:%M")
            print(f"Subject: {mail.subject}, Sender: {mail.sender}, Datetime: {formatted_time}, Read: {mail.read}")

    def open(self, index):
        if index < 0 or index >= len(self.mails):
            print("falscher Index")
            return

        mail = self.mails[index]
        mail.print_full()
        mail.mark_as_read()

    def count_unread(self):
        count = 0
        for mail in self.mails:
            if not mail.read:
                count += 1

        return count


